odoo/fields.py

```diff
@@ -2444,6 +2444,12 @@ def _setup_attrs(self, model_class, name):
                             "field, or change the chosen ondelete policy" % self
                         )
                         continue
+                    if val.startswith('set '):
+                        assert val[4:] in dict(self.selection), (
+                            "%s: ondelete policy of type 'set %%' must be either 'set null', "
+                            "'set default', or 'set value' where value is a valid selection value."
+                        ) % self
+                        continue
                     raise ValueError(
                         "%r: ondelete policy %r for selection value %r is not a valid ondelete "
                         "policy, please choose one of 'set null', 'set default', 'cascade' or "
```

Comment:
complete this message with the new possibility

Reply:
Done, with a more complete commit message